packages/react-positioning/src/usePopper.ts

```diff
@@ -70,222 +45,59 @@ function usePopperOptions(options: PopperOptions, popperOriginalPositionRef: Rea
-            const { x, y } = state.modifiersData.preventOverflow || { x: 0, y: 0 };
-            const { width, height } = state.rects.popper;
-            const basePlacement = getBasePlacement(state.placement);
-
-            const widthProp: keyof PopperJs.SideObject = basePlacement === 'left' ? 'left' : 'right';
-            const heightProp: keyof PopperJs.SideObject = basePlacement === 'top' ? 'top' : 'bottom';
-
-            const applyMaxWidth =
-              autoSize === 'always' ||
-              autoSize === 'width-always' ||
-              (overflow[widthProp] > 0 && (autoSize === true || autoSize === 'width'));
-            const applyMaxHeight =
-              autoSize === 'always' ||
-              autoSize === 'height-always' ||
-              (overflow[heightProp] > 0 && (autoSize === true || autoSize === 'height'));
-
-            if (applyMaxWidth) {
-              state.styles.popper.maxWidth = `${width - overflow[widthProp] - x}px`;
-            }
-            if (applyMaxHeight) {
-              state.styles.popper.maxHeight = `${height - overflow[heightProp] - y}px`;
-            }
-          },
-        },
-
-        /**
-         * This modifier is necessary in order to render the pointer. Refs are resolved in effects, so it can't be
-         * placed under computed modifiers. Deep merge is not required as this modifier has only these properties.
-         */
-        {
-          name: 'arrow',
-          enabled: !!arrow,
-          options: { element: arrow, padding: arrowPadding },
-        },
-
-        /**
-         * Modifies popper offsets to cover the reference rect, but still keep edge alignment
-         */
-        {
-          name: 'coverTarget',
-          enabled: !!coverTarget,
-          phase: 'main',
-          requiresIfExists: ['offset', 'preventOverflow', 'flip'],
-          fn({ state }: PopperJs.ModifierArguments<{}>) {
-            const basePlacement = getBasePlacement(state.placement);
-            switch (basePlacement) {
-              case 'bottom':
-                state.modifiersData.popperOffsets!.y -= state.rects.reference.height;
-                break;
-              case 'top':
-                state.modifiersData.popperOffsets!.y += state.rects.reference.height;
-                break;
-              case 'left':
-                state.modifiersData.popperOffsets!.x += state.rects.reference.width;
-                break;
-              case 'right':
-                state.modifiersData.popperOffsets!.x -= state.rects.reference.width;
-                break;
-            }
-          },
-        },
-      ].filter(Boolean) as PopperJs.Options['modifiers']; // filter boolean conditional spreading values
-
-      const popperOptions: PopperJs.Options = {
-        modifiers,
-
+      const placement = getPlacement(align, position, isRtl);
+      const middleware = [
+        ...[offset && offsetMiddleware(offset)],
```

Comment:
```suggestion
        offset && offsetMiddleware(offset),
```

We don't need to create arrays, correct? If so please update other usages.